scalecube-configuration/src/main/java/io/scalecube/configuration/ConfigurationServiceImpl.java

```diff
@@ -1,169 +1,133 @@
-    }
-  }
-}
+package io.scalecube.configuration;
+
+import io.scalecube.configuration.api.Acknowledgment;
+import io.scalecube.configuration.api.ConfigurationService;
+import io.scalecube.configuration.api.CreateRepositoryRequest;
+import io.scalecube.configuration.api.DeleteRequest;
+import io.scalecube.configuration.api.FetchRequest;
+import io.scalecube.configuration.api.FetchResponse;
+import io.scalecube.configuration.api.SaveRequest;
+import io.scalecube.configuration.authorization.AuthorizationService;
+import io.scalecube.configuration.authorization.OperationType;
+import io.scalecube.configuration.operation.ServiceOperationContext;
+import io.scalecube.configuration.operation.ServiceOperationFactory;
+import io.scalecube.configuration.repository.ConfigurationDataAccess;
+import io.scalecube.configuration.tokens.TokenVerifier;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class ConfigurationServiceImpl implements ConfigurationService {
+
+  private static final Logger logger = LoggerFactory.getLogger(ConfigurationServiceImpl.class);
+
+  private final ConfigurationDataAccess dataAccess;
+  private TokenVerifier tokenVerifier;
+  private final AuthorizationService authorizationService;
+
+  private ConfigurationServiceImpl(
+      ConfigurationDataAccess dataAccess, TokenVerifier tokenVerifier) {
+    this.dataAccess = dataAccess;
+    this.tokenVerifier = tokenVerifier;
+    this.authorizationService = AuthorizationService.builder().build();
+  }
+
+  @Override
+  public Mono<Acknowledgment> createRepository(CreateRepositoryRequest request) {
+    return Mono.fromRunnable(() -> logger.debug("createRepository: enter: request: {}", request))
+        .then(Mono.fromCallable(ServiceOperationFactory::createRepository))
+        .flatMap(operation -> Mono
+            .from(operation.execute(request, context(OperationType.CreateRepoitory))))
+        .subscribeOn(Schedulers.parallel())
+        .doOnSuccess(result -> logger.debug("createRepository: exit: request: {}", request))
+        .doOnError(th -> logger.error("createRepository: request: {}, error: {}", request, th));
+  }
+
+  @Override
+  public Mono<FetchResponse> fetch(FetchRequest request) {
+    return Mono.fromRunnable(() -> logger.debug("fetch: enter: request: {}", request))
+        .then(Mono.fromCallable(ServiceOperationFactory::fetch))
+        .flatMap(operation -> Mono.from(operation.execute(request, context(OperationType.Read))))
+        .subscribeOn(Schedulers.parallel())
+        .doOnSuccess(
+            response -> logger.debug("fetch: exit: request: {}, response: {}", request, response))
+        .doOnError(th -> logger.error("fetch: request: {}, error: {}", request, th));
+  }
+
+  @Override
+  public Flux<FetchResponse> entries(FetchRequest request) {
```

Comment:
how this will work over http?
does it mean u must use ws gw to consume this service?

Reply:
Yep.
Do you think we should return something like Mono<Collection>?

Reply:
i think we should support both.
`public Flux<FetchResponse> entries(FetchRequest request)`

and 

`public Mono<collection> entries(FetchRequest request)`

alternative is to change GW to map to collection in case of flux then wait last and collect

Reply:
See #93